app/process/watcher/prometheus/watcher.go

```diff
@@ -66,27 +68,29 @@ func NewWatcher(
 		assetsMetrics = make(map[uint64]map[string]string)
 	)
 
-	payerAccountBalanceGauge = prometheusService.CreateGaugeIfNotExists(prometheus.GaugeOpts{
-		Name: constants.FeeAccountAmountGaugeName,
-		Help: constants.FeeAccountAmountGaugeHelp,
-		ConstLabels: prometheus.Labels{
-			constants.AccountMetricLabelKey: configuration.Bridge.Hedera.PayerAccount,
-		},
-	})
-	bridgeAccountBalanceGauge = prometheusService.CreateGaugeIfNotExists(prometheus.GaugeOpts{
-		Name: constants.BridgeAccountAmountGaugeName,
-		Help: constants.BridgeAccountAmountGaugeHelp,
-		ConstLabels: prometheus.Labels{
-			constants.AccountMetricLabelKey: configuration.Bridge.Hedera.BridgeAccount,
-		},
-	})
-	operatorBalanceGauge = prometheusService.CreateGaugeIfNotExists(prometheus.GaugeOpts{
-		Name: constants.OperatorAccountAmountName,
-		Help: constants.OperatorAccountAmountHelp,
-		ConstLabels: prometheus.Labels{
-			constants.AccountMetricLabelKey: configuration.Node.Clients.Hedera.Operator.AccountId,
-		},
-	})
+	if prometheusService.GetIsMonitoringEnabled() {
```

Comment:
Why is this needed? The watcher is not registered if this is not enabled.

Reply:
because if to do only 1 check instead of calling each of the methods and to return from inside. It is more efficient like that.

Reply:
@nnedkov-limechain I think @failfmi is referring to the check on init of the Prometheus watcher in [main.go](https://github.com/LimeChain/hashport-validator/blob/feature/dynamic-min-fungible-fees/cmd/main.go#L103)

Reply:
Ouh, yep will remove it.